server/build/webpack.js

```diff
@@ -299,10 +299,11 @@ export default async function createCompiler (dir, { dev = false, quiet = false,
   }
 
   if (!dev) {
+    // We do this to use the minified version of React in production.
+    // This will significant file size redution when turned off uglifyjs.
     webpackConfig.resolve.alias = {
       'react': require.resolve('react/dist/react.min.js'),
-      'react-dom': require.resolve('react-dom/dist/react-dom.min.js'),
-      'react-dom/server': require.resolve('react-dom/dist/react-dom-server.min.js')
```

Comment:
> What do you mean by different entry point?

`react-dom` and `react-dom/server` are different modules. Can verify by the `import` declarations themselves and by the `react-dom/dist/*` directory contents.

>  So, there's no use in that. Isn't it?

There is. It's same reasons to use prod-ready vs dev-mode React in the browser. ReactDOM and `react-dom/server` still has lot of dev-specific code blocks that should be/are properly removed in the minified releases.

The import [here](https://github.com/zeit/next.js/blob/master/server/render.js#L3) will be pulling the dev-mode of `react-dom/server` without this alias. Although Next.js injects a `NODE_ENV` and uglifies, it doesn't do [everything the React team does](https://github.com/facebook/react/blob/master/scripts/rollup/build.js#L168-L215). (Only a subset of prod-build config)